### PR TITLE
Separate functional tests from E2E tests

### PR DESCRIPTION
**Pull request by ecraig12345**
Some of beachball's tests cover individual helper functions, but were being run as part of E2E tests because they need to use fixtures on the filesystem. The problem with this is that if one of the helper functions has a bug, that failure will probably also cause downstream failures in the actual E2E scenario tests, and then logs for the important failure in the helper function will be buried in logs for all the downstream failures.

This PR introduces a third category, functional tests, which are run after unit tests and before E2E tests. I moved some of the existing "E2E" tests into the functional group and added one new test.

NOTE: The files which show as deleted are not actually gone; they're just split into multiple new files.

### CODE REVIEW

File: src/__functional__/changefile/readChangeFiles.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
```
Comment (ecraig12345) [inline src/__functional__/changefile/readChangeFiles.test.ts:1]: This was previously part of `__e2e__/changelog.test.ts`

File: src/__functional__/changefile/writeChangeFiles.test.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import fs from 'fs-extra';
```
Comment (ecraig12345) [inline src/__functional__/changefile/writeChangeFiles.test.ts:1]: This test is new

File: src/__functional__/changelog/writeChangelog.test.ts [hunk 3]
```diff
@@ -0,0 +1,300 @@
+import { generateChangeFiles } from '../../__fixtures__/changeFiles';
```
Comment (ecraig12345) [inline src/__functional__/changelog/writeChangelog.test.ts:1]: This was previously part of `__e2e__/changelog.test.ts`

File: src/__functional__/validation/areChangeFilesDeleted.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import fs from 'fs-extra';
```
Comment (ecraig12345) [inline src/__functional__/validation/areChangeFilesDeleted.test.ts:1]: This was previously part of `__e2e__/validation.test.ts`

File: src/__functional__/validation/isChangeFileNeeded.test.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { defaultRemoteBranchName, defaultRemoteName } from '../../__fixtures__/gitDefaults';
```
Comment (ecraig12345) [inline src/__functional__/validation/isChangeFileNeeded.test.ts:1]: This was previously part of `__e2e__/validation.test.ts`

**Review by kenotron — APPROVED** [review]

**Pull request merged**